src/common/options.rs

```diff
@@ -25,6 +25,13 @@ pub struct Options {
         help = "Path to balena-os image"
     )]
     image: Option<PathBuf>,
+    #[clap(
+        long = "ldd-path",
+        value_name = "LDD_PATH",
+        value_parser,
+        help = "Path to ldd script"
+    )]
+    ldd_path: Option<PathBuf>,
```

Comment:
That's one of these points I think we'll need to discuss in wider audience, but I am personally not a fan of this solution (I chatted a bit with @rahul-thakoor about this, I think he agrees with me). The reasoning is:

1. All our motivation for this work is to enable balenaOS-to-balenaOS migrations.
2. balenaOS does not provide an `ldd`.
3. So, with this solution we are just moving the friction away from us and towards our target users.

I think it would make much more customer-friendly that we solved this ourselves, either in takeover or in the OS.

Reply:
Yes, i'm not convinced we can ask end users to provide their own `ldd` script. I think
- we either ship it in balenaOS and backport where required. customers have to HUP to version that has `ldd` then HUP to next major version
- we find alternatives and do it natively in the rust code. i looked at it briefly a while back, but https://crates.io/crates/lddtree looked promising but would require some changes

Reply:
@lmbarros @rahul-thakoor hmm, in the context of cloud triggered migrations, an approach like this would allow the action script to place the `ldd` script near the `takeover` binary. So I don't think this moves the friction to the target users, because they won't be starting takeover. In the case of a balenaOS to balenaOS migration, that's handled by the action script.

The next best option, apart from the action script handling ldd, would be takeover to either include the script or implement it like you said. But that would be a different effort, which would need to be also maintained. Upstream changes to `ldd` may need to be monitored and maintained, even if they aren't all that frequent.

I'd like to avoid having to tie migration to ldd presence in the hostOS, because that would prevent migration for any device that's not updated to the latest release, which adds friction by requiring an extra step of hostOS update for users.

Reply:
> In the context of cloud triggered migrations, an approach like this would allow the action script to place the ldd script near the takeover binary.

That's a good point, @acostach , which I honestly wasn't keeping in mind! Still, here's a possible counterpoint: I was under the impression that we'll have a period in which users will run takeover manually for balenaOS-to-balenaOS migrations. During this period, the friction would be on their side.

And of course you are right that it would be an additional effort for us to somehow provide `ldd`. We need to weigh this in, too.

> I'd like to avoid having to tie migration to ldd presence in the hostOS, because that would prevent migration for any device that's not updated to the latest release, which adds friction by requiring an extra step of hostOS update for users.

Yup, I get this and it makes sense! I think that would be my current favorite approach, too.

But, food for thought: requiring this extra intermediate HUP could also potentially be a bit safer, because it would force all balenaOS-to-balenaOS migrations to be from an OS release that is at least close to what we tested. Given that this is a risky, not-undoable operation, I sometimes wonder if this extra HUP wouldn't be a good idea.

Reply:
I vote to mark this one as resolved, too. We'll need to revisit the `ldd` issue, but IMO this is not a blocker to the PR.

Reply:
sure, marking as resolved